streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedCogroupedKStreamImpl.java

```diff
@@ -121,11 +121,20 @@
                     + " grace=[" + sessionWindows.gracePeriodMs() + "],"
                     + " retention=[" + retentionPeriod + "]");
             }
-            supplier = Stores.persistentSessionStore(
-                materialized.storeName(),
-                Duration.ofMillis(retentionPeriod)
-            );
+
+            if (materialized.storeType().equals(Materialized.StoreType.IN_MEMORY)) {
```

Comment:
Could we use `switch-case` instead of `if-else` in case we add more store types in the future? Ditto elsewhere.